test/e2e/non_typed_char_error_test.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+package e2e
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestNonTypedCharError(t *testing.T) {
```

Comment:
TestNonTypedArrayFields ? (Might as well want to change the file name and name on line number 49 based on this).

Reply:
updated

Reply:
very nitty, but on a second thought should the name point out that it is testing template rules in this case? (`TestTemplateRulesWithNonTypedArrayFields` maybe, even though it is a mouthful)

Do not feel very strongly about this but I think it does not hurt to be more explicit

Reply:
Agree as this says exactly where the issue was. Let wait for other's opinion on it. @praveenrewar

Reply:
Works for me 👍🏻